### ISSUE TITLE
Flat list option

### ISSUE THREAD

**Issue by borekb** [letmaik/vscode-git-tree-compare#26]
... sort of like the VCS view, to browse the changed files quickly without the need to navigate the full tree.

Awesome extension, by the way!

**Comment by letmaik** [comment]
Thanks!
About your suggestion, why do you find the tree display unsuitable for you? I can imagine that it becomes a bit painful when you have many subfolders, is that the case? How about if subfolders without direct changes (only in their subfolders) were squished into one entry, like:

```
master
  > my/deep/change
    foo.yml
  bar.yml
```
instead of
```
master
  > my
    > deep
      > change
        foo.yml
  bar.yml
```

Would you find that better than a fully flattened list?

**Comment by borekb** [comment]
Hi, and thanks for considering it! Yes, indeed the little usability problem comes from deep trees. The partial flattening that you suggest would help in some cases but not in all, for example, if the change is commonly in two sibling folders. But anything that helps deep trees would be appreciated :)

**Comment by borekb** [comment]
Maybe it doesn't need to be a persistent option but rather a toggle button in the panel header?

**Comment by letmaik** [comment]
Yes, it would definitely be an option as I really like the tree view as well but I agree, it can become cumbersome at times.

What you said about sibling folders, do you mean something like:
```
master
  > my
    > deep
      > change
          foo.yml
      > another
        > change
            bar.yml
```

Wouldn't the following partial flattening work? Or do you find that too confusing?
```
master
  > my/deep
    > change
        foo.yml
    > another/change
        bar.yml
```

Complete flattening would be:
```
master
  > foo.yml - my/deep/change
  > bar.yml - my/deep/another/change
```

For the complete flattening the folder would have to be appended to the file entries, and ideally in that case I would like to replicate the style of the built-in commit view so that the folder name is grayed out or otherwise made less prominent to ease reading/scanning. This level of styling is currently not possible with those custom views however.

**Comment by borekb** [comment]
Incidentally, GitLens added a similar option today in [5.3.0](https://github.com/eamodio/vscode-gitlens/blob/master/CHANGELOG.md#530---2017-09-26), `"gitlens.gitExplorer.files.layout"`.

I took two screenshots of exactly the same screen area, this is the previously default `tree` view:

![image](https://user-images.githubusercontent.com/101152/30901972-0458de0a-a36a-11e7-8cb4-8b24425135ec.png)

This is the flat `list` option:

![image](https://user-images.githubusercontent.com/101152/30901931-e2858d32-a369-11e7-9539-ba36220febeb.png)

Both have something to it:

- Tree gives visual context
- List gives a denser, more information-rich view

Probably switching between the two views quickly would be my preferred way to interact with this.

BTW, your example of partial flattening looks entirely fine to me.

// cc @eamodio

**Comment by letmaik** [comment]
Ah, and it also has partial flattening in the tree view. Good, you convinced me that both views are useful to have. I'll create a separate issue for the partial tree flattening as that is independent of a list view which this issue is about.

**Comment by borekb** [comment]
Thanks, looking forward to :)
